### ISSUE TITLE
Running pin twice will overwrite the comment with the first pinned version

### ISSUE THREAD

**Issue by vallieres** [sethvargo/ratchet#19]
### TL;DR

Running the pin command with a CircleCI config a second time will overwrite the comment that contains the semver text with the hashed version of the first pin execution. Thus loosing the ability to unpin back to the original value.

### Expected behavior

On second pin command, version is detected as already being pinned, thus ignore that line. CircleCI config stays like this:

```yaml
- image: index.docker.io/cimg/go@sha256:0da62eae46657075a5183c115157adb2b40288b9c814ca5a8f9a4291a96a3811 # ratchet:cimg/go:1.18
```

### Observed behavior

The comment will be replaced with the pinned version of the first command.

```yaml
- image: index.docker.io/cimg/go@sha256:0da62eae46657075a5183c115157adb2b40288b9c814ca5a8f9a4291a96a3811 # ratchet:index.docker.io/cimg/go@sha256:0da62eae46657075a5183c115157adb2b40288b9c814ca5a8f9a4291a96a3811
```

### Version

unknown (it's 0.2.1)

### On what operating system are you using ratchet?

Mac OS

### CI/CD workflow file

```yaml
executors:
  default:
    docker:
      - image: cimg/go:1.18
```

### Additional information

Command executed:
`ratchet pin -parser circleci .circleci/config.yml`

**Comment by sethvargo** [comment]
Hi @vallieres 

This is working as intended. If you want to update pins, use the `update` command instead. Pinning an already pinned version should behave exactly as you described it.

**Comment by vallieres** [comment]
I was thinking of integrating Ratchet in our CI pipeline and I was not expecting it to pin the pinned version. When running as part of automated pipelines, it will happen for sure that the pin command will be run more than once.

Which can be a problem since we loose the ability to unpin (and the information about the original version) after a second pin. I understand update can still work once you pinned your dependencies but I believe it should be non-destructive as in if you allow me to pin twice, you should also be able to unpin twice to revert back to your initial state.

ratchet pin -parser circleci .circleci/config.yml (pins the version with the comment)
ratchet pin -parser circleci .circleci/config.yml (pins the pinned version; we lost the original value)
ratchet unpin .circleci/config.yml (restores the first pinned version)
ratchet unpin .circleci/config.yml (does nothing, since we lost the original value)

Think of most database migration tools, if you have 3 versions you can go up 3 time, and down 3 times. I wish Ratchet would work the same way or at the very least once the versions are pinned, not pin them again.

**Comment by sethvargo** [comment]
Hi @vallieres 

Okay - I think it makes sense to ignore already-pinned versions while pinning. I've switched this from a bug to an enchancement.

**Comment by vallieres** [comment]
🙇🏻  thank you! 🙂 Great job by the way! 😄 I really love it!
